### PR TITLE
refactor: improve load with fields projection

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR adds changes suggested in https://github.com/argilla-io/argilla/pull/2540#issuecomment-1470484071.  Also, includes `event_timestamp`  and `status` as minimal required fields.

Refs: #2576  and #2540 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [x] Refactor (change restructuring the codebase without changing functionality)
- [x] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

New tests are included 

**Checklist**

- [x] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [x] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/client/client.py [hunk 1]
```diff
@@ -636,13 +636,21 @@ def _load_records_internal(
         task = dataset.task
 
         task_config = {
-            TaskType.text_classification: (SdkTextClassificationRecord, DatasetForTextClassification),
-            TaskType.token_classification: (SdkTokenClassificationRecord, DatasetForTokenClassification),
-            TaskType.text2text: (SdkText2TextRecord, DatasetForText2Text),
+            TaskType.text_classification: (
+                SdkTextClassificationRecord,
+                DatasetForTextClassification,
+                {"id", "event_timestamp", "status", "inputs", "multi_label"},
```
Comment (davidberenstein1957) [inline src/argilla/client/client.py:642]: should this not also include the "text" parm? or will this be over-ruled by adding "inputs"?

**Comment by davidberenstein1957** [comment]
@frascuchon Intuitively, I agree with having `event_timestamp` and `status` as minimal requirements too, but given the aim of de-coupling of records from their corresponding predictions, maybe `status` ought to be removed? Or will it then always default to `Default`?

File: src/argilla/client/client.py (hunk 1 above)
Comment (frascuchon) [reply src/argilla/client/client.py:642]: For text classification, the `text` param is computed from inputs. Indeed we have a gap between fields for client records and projected `fields` for searches. But this should be tackled in a separate PR.

**Comment by frascuchon** [comment]
> Intuitively, I agree with having `event_timestamp` and `status` as minimal requirements too, but given the aim of de-coupling of records from their corresponding predictions, maybe `status` ought to be removed? Or will it then always default to `Default`?

Since we have default initializers when creating client records providing some default values, if the status is not provided, the default value will be provided. It's weird to have records with a different status value than the store one depending on the field's projection.

Ideally, the load with projection should return a raw bunch of data, instead of a set of records, but the problem here is how to map between the API record model and the client data model for those cases. For example, the  `annotation` and `annotation_agent` are computed fields from the `annotation` field fetched from API.

**Comment by frascuchon** [comment]
Closing in favour of https://github.com/argilla-io/argilla/pull/2668

**Pull request closed without merging**